core/src/main/java/hudson/XmlFile.java

```diff
@@ -185,8 +187,19 @@ private Object unmarshal(Object o, boolean nullOut) throws IOException {
     }
 
     public void write( Object o ) throws IOException {
+        write(o, true);
+    }
+
+    /**
+     * @param force Whether or not to flush the page cache to stable storage with {@link
+     *     FileChannel#force} (i.e., {@code fsync()}} or {@code FlushFileBuffers()}) before this
+     *     method returns. If you set this to {@code false}, you will lose data integrity.
+     */
+    public void write(Object o, boolean force) throws IOException {
         mkdirs();
-        AtomicFileWriter w = new AtomicFileWriter(file);
+        AtomicFileWriter w = force
+                ? new AtomicFileWriter(file)
+                : new AtomicFileWriter(file.toPath(), StandardCharsets.UTF_8, false, false);
```

Comment:
https://github.com/jenkinsci/jenkins/blob/23a941f5b44ff5ef7dc5ef7899d4769f7138f0f0/core/src/main/java/hudson/util/AtomicFileWriter.java#L113-L123 is deprecated. But more OK here? https://github.com/jenkinsci/jenkins/pull/3170 @batmat may have an opinion

Reply:
This seems like an abuse of `@Deprecated`. The method is perfectly OK, but you should think twice about the tradeoffs. I clearly did that.

Reply:
Feel free to remove the annotation if you think it is justifiable to call the method under some circumstances. I agree that `@Deprecated` should not be used merely because you think it is important that a caller read and understand the Javadoc. (case in point: `Run.getAbsoluteUrl`)

Reply:
I'll leave that to Baptiste, since he's the one who introduced this. With a logical change that spans five repositories, I'm trying to minimize the amount of unrelated sprawl.

Reply:
this method should have been restricted and not exposed (but IIRC it was used so we could not at the time)

people have an expectation of atomicFileWriter - it is clearly documented "It will work, or not, but never have work and corrupt stuff".  so now you hare using an AtomicFileWriter in a way that is clearly not going to do that and it is confusing.

https://github.com/jenkinsci/jenkins/blob/23a941f5b44ff5ef7dc5ef7899d4769f7138f0f0/core/src/main/java/hudson/util/AtomicFileWriter.java#L49-L50